### PR TITLE
Code cleanups

### PR DESCRIPTION
**Pull request by andreikoval**

### CODE REVIEW

File: community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnPageCursor.java [hunk 1]
```diff
@@ -796,10 +796,10 @@ public void shiftBytes( int sourceStart, int length, int shift )
         int targetStart = sourceStart + shift;
         int targetEnd = sourceStart + length + shift;
         if ( sourceStart < 0
-                | sourceEnd > filePageSize
-                | targetStart < 0
-                | targetEnd > filePageSize
-                | length < 0 )
```
Comment (chrisvest) [inline community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnPageCursor.java]: Please benchmark these changes.

File: community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/OffHeapPageLock.java [hunk 2]
```diff
@@ -184,7 +184,7 @@ public static boolean tryWriteLock( long address )
             boolean writeCountOverflow = (s & CNT_MASK) == CNT_MASK;
 
             // bitwise-OR to reduce branching and allow more ILP
-            if ( unwritablyLocked | writeCountOverflow )
```
Comment (chrisvest) [inline community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/OffHeapPageLock.java]: Did you see the comment here?

File: community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnPageCursor.java (hunk 1 above)
Comment (andreikoval) [reply community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnPageCursor.java]: Benchmarks are running atm.

File: community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/OffHeapPageLock.java (hunk 2 above)
Comment (andreikoval) [reply community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/OffHeapPageLock.java]: Sorry, missed that one. I can revert it after benchmarks finish.

File: community/kernel/src/main/java/org/neo4j/kernel/impl/cache/MeasureDoNothing.java [hunk 3]
```diff
@@ -21,6 +21,10 @@
 
 import org.neo4j.logging.Log;
 
+/**
+ * @deprecated Remove this nonsense and use standard beans for getting GC measurements
+ */
+@Deprecated
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/cache/MeasureDoNothing.java:27]: It's measuring all kinds of stalls, not just stalls caused by GC. An interesting case is for instance documented at: http://www.evanjones.ca/jvm-mmap-pause.html
I don't think this is public API, though, so you can remove it without deprecating.

File: enterprise/kernel/src/main/java/org/neo4j/kernel/impl/store/format/highlimit/v300/BaseHighLimitRecordFormatV3_0_0.java [hunk 4]
```diff
@@ -120,7 +120,8 @@ public void read( RECORD record, PageCursor primaryCursor, RecordLoad mode, int
             long pageId = pageIdForRecord( secondaryId, primaryCursor.getCurrentPageSize(), recordSize );
             int offset = offsetForId( secondaryId, primaryCursor.getCurrentPageSize(), recordSize );
             PageCursor secondaryCursor = primaryCursor.openLinkedCursor( pageId );
-            if ( (!secondaryCursor.next()) | offset < 0 )
+            final boolean secondaryHasNext = secondaryCursor.next();
```
Comment (MishaDemianenko) [inline enterprise/kernel/src/main/java/org/neo4j/kernel/impl/store/format/highlimit/v300/BaseHighLimitRecordFormatV3_0_0.java]: please don't touch old formats, whatever they do let them do that

File: enterprise/kernel/src/main/java/org/neo4j/kernel/impl/store/format/highlimit/v300/PropertyRecordFormatV3_0_0.java [hunk 5]
```diff
@@ -76,7 +76,7 @@ public void read( PropertyRecord record, PageCursor cursor, RecordLoad mode, int
             record.initialize( inUse,
                     toAbsolute( Reference.decode( cursor ), recordId ),
                     toAbsolute( Reference.decode( cursor ), recordId ) );
-            if ( (blockCount > record.getBlockCapacity()) | (RECORD_SIZE - (cursor.getOffset() - offset) < blockCount * Long.BYTES) )
+            if ( (blockCount > record.getBlockCapacity()) || (RECORD_SIZE - (cursor.getOffset() - offset) < blockCount * Long.BYTES) )
```
Comment (MishaDemianenko) [inline enterprise/kernel/src/main/java/org/neo4j/kernel/impl/store/format/highlimit/v300/PropertyRecordFormatV3_0_0.java]: please don't touch old formats, whatever they do let them do that

File: enterprise/kernel/src/main/java/org/neo4j/kernel/impl/store/format/highlimit/v306/BaseHighLimitRecordFormatV3_0_6.java [hunk 6]
```diff
@@ -122,7 +122,8 @@ public void read( RECORD record, PageCursor primaryCursor, RecordLoad mode, int
             long pageId = pageIdForRecord( secondaryId, primaryCursor.getCurrentPageSize(), recordSize );
             int offset = offsetForId( secondaryId, primaryCursor.getCurrentPageSize(), recordSize );
             PageCursor secondaryCursor = primaryCursor.openLinkedCursor( pageId );
-            if ( (!secondaryCursor.next()) | offset < 0 )
+            final boolean secondaryHasNext = secondaryCursor.next();
```
Comment (MishaDemianenko) [inline enterprise/kernel/src/main/java/org/neo4j/kernel/impl/store/format/highlimit/v306/BaseHighLimitRecordFormatV3_0_6.java]: please don't touch old formats, whatever they do let them do that

File: enterprise/kernel/src/main/java/org/neo4j/kernel/impl/store/format/highlimit/v310/BaseHighLimitRecordFormatV3_1_0.java [hunk 7]
```diff
@@ -133,7 +133,8 @@ public void read( RECORD record, PageCursor primaryCursor, RecordLoad mode, int
             long pageId = pageIdForRecord( secondaryId, primaryCursor.getCurrentPageSize(), recordSize );
             int offset = offsetForId( secondaryId, primaryCursor.getCurrentPageSize(), recordSize );
             PageCursor secondaryCursor = primaryCursor.openLinkedCursor( pageId );
-            if ( (!secondaryCursor.next()) | offset < 0 )
+            final boolean secondaryHasNext = secondaryCursor.next();
```
Comment (MishaDemianenko) [inline enterprise/kernel/src/main/java/org/neo4j/kernel/impl/store/format/highlimit/v310/BaseHighLimitRecordFormatV3_1_0.java]: please don't touch old formats, whatever they do let them do that

File: enterprise/kernel/src/main/java/org/neo4j/kernel/impl/store/format/highlimit/v306/PropertyRecordFormatV3_0_6.java [hunk 8]
```diff
@@ -114,7 +114,7 @@ public void read( PropertyRecord record, PageCursor cursor, RecordLoad mode, int
                         toAbsolute( Reference.decode( cursor ), recordId ) );
             }
             record.setUseFixedReferences( useFixedReferences );
-            if ( (blockCount > record.getBlockCapacity()) | (RECORD_SIZE - (cursor.getOffset() - offset) < blockCount * Long.BYTES) )
+            if ( (blockCount > record.getBlockCapacity()) || (RECORD_SIZE - (cursor.getOffset() - offset) < blockCount * Long.BYTES) )
```
Comment (MishaDemianenko) [inline enterprise/kernel/src/main/java/org/neo4j/kernel/impl/store/format/highlimit/v306/PropertyRecordFormatV3_0_6.java]: please don't touch old formats, whatever they do let them do that

File: enterprise/kernel/src/main/java/org/neo4j/kernel/impl/store/format/highlimit/v310/PropertyRecordFormatV3_1_0.java [hunk 9]
```diff
@@ -115,7 +115,7 @@ public void read( PropertyRecord record, PageCursor cursor, RecordLoad mode, int
                         toAbsolute( Reference.decode( cursor ), recordId ) );
             }
             record.setUseFixedReferences( useFixedReferences );
-            if ( (blockCount > record.getBlockCapacity()) | (RECORD_SIZE - (cursor.getOffset() - offset) < blockCount * Long.BYTES) )
+            if ( (blockCount > record.getBlockCapacity()) || (RECORD_SIZE - (cursor.getOffset() - offset) < blockCount * Long.BYTES) )
```
Comment (MishaDemianenko) [inline enterprise/kernel/src/main/java/org/neo4j/kernel/impl/store/format/highlimit/v310/PropertyRecordFormatV3_1_0.java]: please don't touch old formats, whatever they do let them do that

File: enterprise/kernel/src/main/java/org/neo4j/kernel/impl/store/format/highlimit/v320/BaseHighLimitRecordFormatV3_2_0.java [hunk 10]
```diff
@@ -133,7 +133,8 @@ public void read( RECORD record, PageCursor primaryCursor, RecordLoad mode, int
             long pageId = pageIdForRecord( secondaryId, primaryCursor.getCurrentPageSize(), recordSize );
             int offset = offsetForId( secondaryId, primaryCursor.getCurrentPageSize(), recordSize );
             PageCursor secondaryCursor = primaryCursor.openLinkedCursor( pageId );
-            if ( (!secondaryCursor.next()) | offset < 0 )
+            final boolean secondaryHasNext = secondaryCursor.next();
```
Comment (MishaDemianenko) [inline enterprise/kernel/src/main/java/org/neo4j/kernel/impl/store/format/highlimit/v320/BaseHighLimitRecordFormatV3_2_0.java]: please don't touch old formats, whatever they do let them do that

File: enterprise/kernel/src/main/java/org/neo4j/kernel/impl/store/format/highlimit/v320/PropertyRecordFormatV3_2_0.java [hunk 11]
```diff
@@ -115,7 +115,7 @@ public void read( PropertyRecord record, PageCursor cursor, RecordLoad mode, int
                         toAbsolute( Reference.decode( cursor ), recordId ) );
             }
             record.setUseFixedReferences( useFixedReferences );
-            if ( (blockCount > record.getBlockCapacity()) | (RECORD_SIZE - (cursor.getOffset() - offset) < blockCount * Long.BYTES) )
+            if ( (blockCount > record.getBlockCapacity()) || (RECORD_SIZE - (cursor.getOffset() - offset) < blockCount * Long.BYTES) )
```
Comment (MishaDemianenko) [inline enterprise/kernel/src/main/java/org/neo4j/kernel/impl/store/format/highlimit/v320/PropertyRecordFormatV3_2_0.java]: please don't touch old formats, whatever they do let them do that

**Review by MishaDemianenko — CHANGES_REQUESTED** [review]
Please revert old format changes, otherwise, do not have any more comments. Just make CV happy with benchmarks results

File: community/kernel/src/main/java/org/neo4j/kernel/impl/cache/MeasureDoNothing.java (hunk 3 above)
Comment (MishaDemianenko) [reply community/kernel/src/main/java/org/neo4j/kernel/impl/cache/MeasureDoNothing.java:27]: This thing will be replaced by 2 things: one that reports real gc pauses and another one that reports stalls.

Comment (andreikoval) [reply community/kernel/src/main/java/org/neo4j/kernel/impl/cache/MeasureDoNothing.java:27]: This one is replaced with #11010

**Comment by andreikoval** [comment]
@chrisvest I've reverted those short-circuit boolean expr changes due to inconclusive benchmarks result.

**Review by MishaDemianenko — APPROVED** [review]

**Pull request merged**